python/ray/data/_internal/delegating_block_builder.py

```diff
@@ -56,13 +56,28 @@ def will_build_yield_copy(self) -> bool:
 
     def build(self) -> Block:
         if self._builder is None:
-            if self._empty_block is not None:
-                self._builder = BlockAccessor.for_block(self._empty_block).builder()
-                self._builder.add_block(self._empty_block)
-            else:
-                self._builder = ArrowBlockBuilder()
+            return self.generate_empty_block(self._empty_block)
         return self._builder.build()
 
+    def generate_empty_block(self, sample_block: Optional[Block] = None) -> Block:
```

Comment:
what's the difference between this method and 

```py
if sample_block is not None:
  builder.add(sample_block)
builder.build()
```

I feel this method is not necessarilly needed.

Reply:
The abstractions here are very messy so I wanted to make this more clear...but can remove for now.

Reply:
updated